rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -894,6 +897,26 @@ stream {
         location = {{ $authPath }} {
             internal;
 
+            {{ if $externalAuth.AuthCacheKey }}
+            set $cache_key '{{ $externalAuth.AuthCacheKey }}';
+
+            rewrite_by_lua_block {
+                local hash = ngx.sha1_bin(ngx.var.cache_key)
+                ngx.var.cache_key = ngx.encode_base64(hash)
+            }
+
+            proxy_cache auth_cache;
+
+            {{ if not $externalAuth.AuthCacheDuration }}
+            proxy_cache_valid 200 202 401 5m;
```

Comment:
Can you make sure this is handled in the code, i.e by default `AuthCacheDuration` has this value. It's always better to avoid logic in template.

Reply:
I centralized it in https://github.com/kubernetes/ingress-nginx/pull/4278/commits/d6b901419501b9bf711d650ac7dd0d9bee05b180.